frontstage/views/surveys.py

```diff
@@ -1,37 +1,164 @@
+
+    return render_template('surveys/surveys-todo.html', data=template_data,
+                           _theme='default', sorted_surveys_list=sorted_surveys_list)
 
 
 @surveys_bp.route('/history', methods=['GET'])
 @jwt_authorization(request)
 def surveys_history(session):
     party_id = session['party_id']
     surveys_list = get_surveys_list(party_id, 'history')
-    return render_template('surveys/surveys-history.html',  _theme='default',
+    template_data = {"survey": {"justAdded": "false"}}
+    return render_template('surveys/surveys-history.html',  _theme='default', data=template_data,
                            surveys_list=surveys_list, history=True)
 
 
+@surveys_bp.route('/add-survey', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def add_survey(session):
+    form = EnrolmentCodeForm(request.form)
+
+    if request.method == 'POST' and form.validate():
+        logger.info('Enrolment code submitted')
+        enrolment_code = request.form.get('enrolment_code').lower()
+        request_data = {
+            'enrolment_code': enrolment_code,
+            'initial': True
+        }
+        response = api_call('POST', app.config['VALIDATE_ENROLMENT'], json=request_data)
+
+        # Handle API errors
+        if response.status_code == 404:
+            logger.info('Enrolment code not found')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
+                                   form=form, data=template_data), 202
+        elif response.status_code == 401 and not json.loads(response.text).get('active'):
+            logger.info('Enrolment code not active')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
+                                   form=form, data=template_data), 200
+        elif response.status_code == 400:
+            logger.info('Enrolment code already used')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
+                                   form=form, data=template_data), 200
+        elif response.status_code != 200:
+            logger.error('Failed to submit enrolment code')
+            raise ApiError(response)
+
+        encrypted_enrolment_code = cryptographer.encrypt(enrolment_code.encode()).decode()
+        logger.info('Successful enrolment code submitted')
+        return redirect(url_for('surveys_bp.survey_confirm_organisation',
+                                encrypted_enrolment_code=encrypted_enrolment_code,
+                                _external=True,
+                                _scheme=getenv('SCHEME', 'http')))
+
+    elif request.method == 'POST' and not form.validate():
+        logger.info('Invalid character length, must be 12 characters')
+        template_data = {"error": {"type": "failed"}}
```

Comment:
We don't need to define template_data here, just pass the json into the template directly, i.e.
`return render_template('surveys/surveys-add.html', _theme='default', form=form, data={"error": {"type": "failed"}})`